src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java

```diff
@@ -79,28 +85,52 @@ public StreamingClientProperties(Map<String, String> connectorConfig) {
         STREAMING_CLIENT_PREFIX_NAME
             + connectorConfig.getOrDefault(Utils.NAME, DEFAULT_CLIENT_NAME);
 
-    // Override only if the max client lag is explicitly set in config
+    // Override only if the streaming client property is explicitly set in config
     this.parameterOverrides = new HashMap<>();
     Optional<String> snowpipeStreamingMaxClientLag =
         Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_MAX_CLIENT_LAG));
     snowpipeStreamingMaxClientLag.ifPresent(
-        overriddenValue -> {
-          parameterOverrides.put(MAX_CLIENT_LAG, String.format("%s second", overriddenValue));
-        });
+        overriddenValue ->
+            parameterOverrides.put(MAX_CLIENT_LAG, String.format("%s second", overriddenValue)));
+
+    if (isSingleBufferEnabled(connectorConfig)) {
```

Comment:
question for my understanding - why are we only enabling the max channel and max memory size for the single buffer?

Reply:
First thing: isSingleBufferEnabled serves as parameter protection, we don't want to change any logic when the feature is disabled.

Second thing: these properties should only be available when using single buffer, the customer can still change the settings using overrideMap as it was before for multiple buffers but we don't want to explicitly expose them for (it should be somehow hidden)